birdset/modules/models/ssast.py

```diff
@@ -0,0 +1,462 @@
+import torch.nn as nn
+import torch
+from timm.models.layers import trunc_normal_
+import timm
+import numpy as np
+from timm.models.layers import to_2tuple
+from random import randrange
+import random
+from typing import Tuple
+
+# override the timm package to relax the input shape constraint.
+class PatchEmbed(nn.Module):
+    """ Image to Patch Embedding
+    """
+    def __init__(self, img_size=224, patch_size=16, in_chans=3, embed_dim=768):
+        super().__init__()
+        img_size = to_2tuple(img_size)
+        patch_size = to_2tuple(patch_size)
+        num_patches = (img_size[1] // patch_size[1]) * (img_size[0] // patch_size[0])
+        self.img_size = img_size
+        self.patch_size = patch_size
+        self.num_patches = num_patches
+
+        self.proj = nn.Conv2d(in_chans, embed_dim, kernel_size=patch_size, stride=patch_size)
+
+    def forward(self, x):
+        x = self.proj(x).flatten(2).transpose(1, 2)
+        return x
+
+class ASTModel(nn.Module):
+    def __init__(self, label_dim=527,
```

Comment:
@XgamerTV This `_init_` function is quite long. Could you split it up into smaller functions to improve readability?

Reply:
I removed unneeded code and explained it more